### ISSUE TITLE
Container ImagePullPolicy builds an incompatible type

### ISSUE THREAD

**Issue by tcjennings** [argoproj-labs/hera#888]
<h2>Pre-bug-report checklist</h2>

**1. This bug can be reproduced using pure Argo YAML**
- [ ] Yes [👉 Please report a bug to the Argo Workflows GitHub 👈](https://github.com/argoproj/argo-workflows/issues/new/choose)
- [X] No

_If yes, it is more likely to be an Argo bug unrelated to Hera. Please double check before submitting an issue to Hera._

**2. This bug occurs in Hera when...**
- [X] exporting to YAML
- [ ] submitting to Argo
- [ ] running on Argo with the Hera runner
- [X] other: creating a workflow

<h2>Bug report</h2>

**Describe the bug**
When applying an `image_pull_policy` to a `Container`, the resulting attribute is not compatible with the pydantic model. The model expects a string, but the `_build_image_pull_policy` returns an `enum` type instead. The same function in the `UserContainer` model works correctly.

This seems to have been a bug since 5.5.0.

_Error log if applicable_:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/x/Library/Caches/pypoetry/virtualenvs/hera-bug-z6maGZWj-py3.11/lib/python3.11/site-packages/hera/workflows/workflow.py", line 369, in to_yaml
    return _yaml.dump(self.to_dict(), *args, **kwargs)
                      ^^^^^^^^^^^^^^
  File "/Users/xLibrary/Caches/pypoetry/virtualenvs/hera-bug-z6maGZWj-py3.11/lib/python3.11/site-packages/hera/workflows/workflow.py", line 353, in to_dict
    return self.build().dict(exclude_none=True, by_alias=True)
           ^^^^^^^^^^^^
  File "/Users/x/Library/Caches/pypoetry/virtualenvs/hera-bug-z6maGZWj-py3.11/lib/python3.11/site-packages/hera/workflows/workflow.py", line 349, in build
    return _WorkflowModelMapper.build_model(Workflow, self, model_workflow)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/x/Library/Caches/pypoetry/virtualenvs/hera-bug-z6maGZWj-py3.11/lib/python3.11/site-packages/hera/workflows/_mixins.py", line 1217, in build_model
    getattr(hera_obj, mapper.builder.__name__)()
  File "/Users/x/Library/Caches/pypoetry/virtualenvs/hera-bug-z6maGZWj-py3.11/lib/python3.11/site-packages/hera/workflows/workflow.py", line 131, in _build_templates
    templates.append(template._build_template())
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/xLibrary/Caches/pypoetry/virtualenvs/hera-bug-z6maGZWj-py3.11/lib/python3.11/site-packages/hera/workflows/container.py", line 80, in _build_template
    container=self._build_container(),
              ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/x/Library/Caches/pypoetry/virtualenvs/hera-bug-z6maGZWj-py3.11/lib/python3.11/site-packages/hera/workflows/container.py", line 49, in _build_container
    return _ModelContainer(
           ^^^^^^^^^^^^^^^^
  File "pydantic/main.py", line 341, in pydantic.main.BaseModel.__init__
pydantic.error_wrappers.ValidationError: 1 validation error for Container
imagePullPolicy
  str type expected (type=type_error.str)
```

**To Reproduce**
Full Hera code to reproduce the bug:
```py
from hera.workflows import Workflow, Container, UserContainer
from hera.workflows.models import ImagePullPolicy

# Using the enum
with Workflow(name="my-workflow", namespace="argo") as w:
    Container(image_pull_policy=ImagePullPolicy.always)

w.to_yaml()
# !!error

# Using a string
with Workflow(name="my-workflow", namespace="argo") as w:
    Container(image_pull_policy="Always")

w.create()
# !!error

# Applying the ImagePullPolicy to a sidecar container instead:
with Workflow(name="my-workflow", namespace="argo") as w:
    Container(name="main",
        sidecars=UserContainer(name="sidecar", image="python:3", image_pull_policy=ImagePullPolicy.always)
    )

w.to_yaml()
# no error
```

**Expected behavior**
The workflow should be built successfully with the specified imagePullPolicy whether it was provided as an enum or a string as allowed by the `Container` class.

**Environment**
- Hera Version: 5.10.1
- Python Version: 3.11.6
- Argo Version: NA

**Comment by elliotgunton** [comment]
Looks related to #730 and #797, which should have been fixed by #798 cc @flaviuvadan

**Comment by tcjennings** [comment]
I don't think #798 fixed this for `Container` and I see a review comment from @flaviuvadan acknowledging there are different `_build_image_pull_policy` methods that have differing return values.
